### PR TITLE
Show the user a URL when they report a known problem

### PR DESCRIPTION
**Pull request by hughsie**
Resolves half of https://github.com/hughsie/lvfs-website/issues/43

### CODE REVIEW

File: src/fu-util.c [hunk 1]
```diff
@@ -727,6 +732,62 @@ fu_util_report_history_for_uri (FuUtilPrivate *priv,
 				  SOUP_MEMORY_COPY, data, strlen (data));
 	status_code = soup_session_send_message (priv->soup_session, msg);
 	g_debug ("server returned: %s", msg->response_body->data);
```
Comment (superm1) [inline src/fu-util.c]: Since you're actually parsing the response now, probably don't need this debug line anymore.

File: src/fu-util.c [hunk 2]
```diff
@@ -727,6 +732,62 @@ fu_util_report_history_for_uri (FuUtilPrivate *priv,
...
+
+	/* parse JSON reply */
+	json_parser = json_parser_new ();
+	if (!json_parser_load_from_data (json_parser,
+					 msg->response_body->data,
+					 msg->response_body->length,
+					 error)) {
+		g_prefix_error (error, "Failed to parse JSON response: ");
```
Comment (superm1) [inline src/fu-util.c]: Not familiar enough with `json_parser_load_from_data` to know, but if it doesn't show what the response it was  trying to parse into the `GError`, probably should add it here.

File: src/fu-util.c [hunk 3]
```diff
@@ -727,6 +732,62 @@ fu_util_report_history_for_uri (FuUtilPrivate *priv,
...
+		return FALSE;
+	}
+	json_root = json_parser_get_root (json_parser);
+	if (json_root == NULL) {
+		g_set_error_literal (error,
+				     FWUPD_ERROR,
+				     FWUPD_ERROR_PERMISSION_DENIED,
+				     "JSON response was malformed");
```
Comment (superm1) [inline src/fu-util.c]: `: %s, msg->response_body->data`

File: src/fu-util.c (hunk 1 above)
Comment (hughsie) [reply src/fu-util.c]: I'd be a lot more comfortable leaving it in; it allows us to debug user problems without using gdb.

File: src/fu-util.c [hunk 4]
```diff
@@ -727,6 +732,62 @@ fu_util_report_history_for_uri (FuUtilPrivate *priv,
...
+	}
+
+	/* server wanted us to see the message */
+	if (server_msg != NULL) {
+		if (g_strcmp0 (server_msg, "known issue") == 0 &&
+		    json_object_has_member (json_object, "uri")) {
+			g_print ("%s %s\n",
+				 /* TRANSLATORS: the server sent the user a small message */
+				 _("Update failure is a known issue, visit this URL for more information:"),
+				 json_object_get_string_member (json_object, "uri"));
```
Comment (superm1) [inline src/fu-util.c]: Perhaps offer to xdg-open it at this point with a prompt?

File: src/fu-util.c (hunk 2 above)
Comment (hughsie) [reply src/fu-util.c]: Fixed.

File: src/fu-util.c (hunk 1 above)
Comment (superm1) [reply src/fu-util.c]: OK

File: src/fu-util.c (hunk 3 above)
Comment (hughsie) [reply src/fu-util.c]: Fixed.

File: src/fu-util.c (hunk 4 above)
Comment (hughsie) [reply src/fu-util.c]: I did ponder that; I think the user wouldn't expect that from a console program. From gnome-software, sure, just not here.

Comment (superm1) [reply src/fu-util.c]: Gotcha, makes sense.

**Pull request merged**